source/test-suite/Main.hs

```diff
@@ -2439,8 +2443,39 @@ spec = describe "Witch" $ do
       it "works" $ do
         f "a" `shouldBe` Tagged.Tagged (LazyByteString.pack [0x00, 0x00, 0x00, 0x61])
 
+  describe "Generically" $ do
+    it "converts into pair" $ do
+      let f = Witch.from @(Int, Bool) @(Pair Int Bool)
+      f (0, False) `shouldBe` MkPair 0 False
+
+    it "converts into pair while also converting fields" $ do
+      let f = Witch.from @(Int, Int.Int8) @(Pair Integer Age)
+      f (1, 2) `shouldBe` MkPair 1 (MkAge 2)
+
+    it "converts from pair" $ do
+      let f = Witch.from @(Pair Int Bool) @(Int, Bool)
+      f (MkPair 0 False) `shouldBe` (0, False)
+
+    it "converts from pair while also converting fields" $ do
+      let f = Witch.from @(Pair Int Age) @(Integer, Int.Int8)
+      f (MkPair 1 (MkAge 2)) `shouldBe` (1, 2)
+
+data Pair a b
+  = MkPair a b
+  deriving (Eq, Generics.Generic, Show)
+
+deriving via
+  Generics.Generically (Pair c d)
+  instance
+    (Witch.From a c, Witch.From b d) => Witch.From (a, b) (Pair c d)
```

Comment:
This shows what typical usage would be like. I only test pairs here because they're convenient, but this should work with anything. I should probably add tests for the following:

``` hs
-- V1
data Empty
-- instance From Empty Void

-- U1
data Unit = Unit
-- instance From Unit ()

-- K1
newtype Only a = MkOnly a
-- instance From a b => From (Only a) (Solo a)

-- (:+:) already covered by pair

-- (:*:)
data Result a b = Err a | Ok b
-- instance From a b => From (Result a b) (Either a b)

-- not sure if this is a separate Rep constructor or not
-- but it would be good to test either way
data List a = Nil | Cons a (List a)
-- instance From a b => From (List a) [a]
```